### PR TITLE
Use safe random instead of deprecated rand() function

### PR DESCRIPTION
**Pull request by willemstuursma**
The php function `rand()` is not quite safe and not even very random. This PR pulls in the polyfill `paragonie/random_compat` for the PHP7 function `random_int()` and uses this function for getting a random key. 

This deprecates the `seedRandom()` method.

### CODE REVIEW

**Comment by malkusch** [comment]
Could you please elaborate on the issues you had with `rand()`? Did you experience collisions? The token doesn't need any cryptographic properties.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/vendor/
```
Comment (aequasi) [inline .gitignore]: This should probably be in your global gitignore

**Comment by willemstuursma** [comment]
Hi @malkusch. Thanks for your reply. We are in the process of migrating from MySQL named locks to Redis locking. We are not experiencing any issues yet, I'm just worried. The `seedRandom()` method requires me to investigate if we use fork based Apache, or thread based, etc. to be sure that the distribution will always works correctly in our set up. And it requires me to keep track of this method, if we ever were to switch http server.

Switching to `random_int()` would assure me that this distribution would always be uniform, irrespective of the underlying platform.

**Comment by malkusch** [comment]
No you don't have to worry. That use case is safe. My comment on seeding is about the rare case were forking happens after the Lock object was created.  This might happen in some rare esotheric CLI scripts. As long as you don't have an actual issue, I'm not going to merge this PR. It brings accidential complexity without any benefit.

**Comment by yuloh** [comment]
> As long as you don't have an actual issue, I'm not going to merge this PR.

I'm 99% certain I was experiencing this issue.  It's pretty difficult to reproduce because it's random 😀 , but after seeding `rand` with `random_int` I stopped experiencing issues.  This was with 1-20 non-forking processes attempting to use the same lock.  As outlined in #9 the mutex regenerating the token every time it fails to acquire a locks seems to contribute to the likely hood of causing a collision.

**Pull request closed without merging**